libs/database/src/lib/entities/delayed-job.entity.ts

```diff
@@ -44,5 +61,29 @@ export class DelayedJob extends Model<DelayedJob> {
 
   @Column(BOOLEAN)
   isAcknowledged: boolean;
-  
+
+  @AllowNull
+  @Column(STRING)
+  name: string | null;
+
+  @AllowNull
+  @Column(STRING)
+  entityType: string | null;
+
+  @AllowNull
+  @Column(BIGINT.UNSIGNED)
+  entityId: number | null;
+
+  /**
+   * Fetch the related entity dynamically based on the entityType and entityId.
+   */
+  async getRelatedEntity() {
+    const Model = ENTITY_TYPE_MAPPING[this.entityType || ""];
+    if (Model && this.entityId) {
+      const entity = await Model.findByPk(this.entityId);
+      return entity?.name;
+    }
+
+    return null;
+  }
```

Comment:
This related entity system should be using scoped `@BelongsTo` associations (see tree-species.entity.ts for an example). With that in place, it should be possible to update the query in the controller to fetch the associated record at the same time as the jobs by including the association. Let me know if you have trouble wiring that up; I've spent a fair mount of time messing around with those systems at this point.

Reply:
changed to belong to